src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let resultSum = 0;
+
+  const funAdder = (a) => {
+    if (a === undefined) {
+      const returnValue = resultSum;
```

Comment:
Storing the result in a separate variable before resetting `resultSum` is a good approach to ensure that the correct value is returned even if `funAdder` is called again in the future.